classes/services/PKPFileService.inc.php

```diff
@@ -181,9 +181,10 @@ public function download($fileId, $filename, $inline = false)
      */
     public function formatFilename($path, $filename)
     {
-        $extension = strtolower(pathinfo($path, PATHINFO_EXTENSION));
+        preg_match("/(\\.\\w{1,3})?\\.\\w+$/", $path, $extension); #extended matching which also captures .tar.gz extensions
+        $extension = substr($extension[0], 1); # Remove leading dot
```

Comment:
This will cause a warning if there is no file extension. I'd suggest checking the return value of `preg_match` before using `$extension[0]`.

Reply:
@asmecher Thanks for the suggestion, I refactored the code as well. The warning should have been taken care of and the code is now more readable.